torch/distributions/constraints.py

```diff
@@ -1,3 +1,22 @@
+r"""
+The following constraints are implemented:
+
+- ``constraints.boolean``
+- ``constraints.dependent``
+- ``constraints.greater_than(lower_bound)``
+- ``constraints.integer_interval(lower_bound, upper_bound)``
+- ``constraints.interval(lower_bound, upper_bound)``
+- ``constraints.lower_cholesky``
+- ``constraints.lower_triangular``
+- ``constraints.lower_triangular``
+- ``constraints.nonnegative_integer``
+- ``constraints.positive``
+- ``constraints.positive_integer``
+- ``constraints.real``
+- ``constraints.simplex``
+- ``constraints.unit_interval``
+"""
```

Comment:
Is there any reason why we don't want to put the registry code inside the `constraints` file? It would seem logical to do `from torch.distributions.constraints import biject_to`

Reply:
The registry depends on `constraints` and `transforms`, and `transforms` depends on `constraints`. Therefore if we were to move the registry to an existing file, it would need to live at the bottom of `transforms.py`. But that file is already crowded and will only grow in the future. Therefore I opted for a new separate file.

Reply:
Right. I still think `form torch.distributions.constraint_registry import biject_to` is kind of ugly (registry is an implementation detail). How about importing those two functions at the bottom of `transforms.py`?

Reply:
Note that those two functions are in `torch.distributions`, so you can succinctly
```py
from torch.distributions import biject_to, transform_to
```
I'm not sure we could cleanly import them into transforms.py.

Reply:
Alright, that's good! Thanks